### PR TITLE
Add SHA256 and SHA512 based output for 'openssl passwd'

### PR DESCRIPTION
**Pull request by levitte**
[RT#4674](https://rt.openssl.org/Ticket/Display.html?id=4674) issue 2

### CODE REVIEW

**Comment by richsalz** [comment]
should you add a test based on the selftest vectors from ulrich's page?

File: apps/passwd.c [hunk 1]
```diff
@@ -43,13 +50,13 @@ static unsigned const char cov_2char[64] = {
 static int do_passwd(int passed_salt, char **salt_p, char **salt_malloc_p,
                      char *passwd, BIO *out, int quiet, int table,
                      int reverse, size_t pw_maxlen, int usecrypt, int use1,
-                     int useapr1);
+                     int useapr1, int use5, int use6);
```
Comment (richsalz) [inline apps/passwd.c]: i'd really prefer a single 'mode' variable, rather than adding a new flag param, since only one flag can be set at a time ...

**Comment by levitte** [comment]
> should you add a test based on the selftest vectors from ulrich's page?

Isn't that exactly what I've added to `20-test_passwd.t`?  Or do you mean the FIPS test vectors you see in his code?  That's for SHA256 and SHA512 themselves, and we might already have that covered elsewhere.  Either way, that's got nothing to do with this PR.

File: apps/passwd.c (hunk 1 above)
Comment (levitte) [inline apps/passwd.c]: Sure

**Comment by richsalz** [comment]
Sorry, I missed the test-passwd stuff.  Must have blinked. :)

**Comment by levitte** [comment]
Ok, modification to a `mode` variable rather than `use1`, `usecrypt`, etc done.

**Comment by richsalz** [comment]
+1 ship

**Comment by levitte** [comment]
Thank you!  Merged.

**Pull request closed without merging**